### ISSUE TITLE
TraversableOnce.max with NaN should return NaN

### ISSUE THREAD

**Issue by scabug** [scala/bug#9637]
scala> Seq(1.0, Double.NaN, 2.5E-4).max
res2: Double = 2.5E-4

From a user point of view it would be great if this operation could
return NaN.

**Comment by som-snytt** [comment]
```
scala 2.13.0-M5> Seq(1.0, Double.NaN, 2.5E-4).max
                                              ^
                 warning: object DeprecatedDoubleOrdering in object Ordering is deprecated (since 2.13.0): There are multiple ways to order Doubles (Ordering.Double.TotalOrdering, Ordering.Double.IeeeOrdering). Specify one by using a local import, assigning an implicit val, or passing it explicitly. See the documentation for details.
res0: Double = NaN
```

**Comment by NthPortal** [comment]
Fixed by scala/scala#6410
